joblib/test/test_numpy_pickle.py

```diff
@@ -156,21 +153,35 @@ def test_numpy_persistence():
     filename = env['filename']
     rnd = np.random.RandomState(0)
     a = rnd.random_sample((10, 2))
-    for compress, cache_size in ((0, 0), (1, 0), (1, 10)):
+    for compress, cache_size in ((0, None), (0, None),
```

Comment:
No need to check different cache_size values anymore, just loop over compress.

I would put the warnings test if cache_size is set in a different test function for simplicity.